x-pack/plugin/downsample/src/internalClusterTest/java/org/elasticsearch/xpack/downsample/DataStreamLifecycleDownsampleIT.java

```diff
@@ -277,5 +269,111 @@ public void testUpdateDownsampleRound() throws Exception {
             assertThat(writeIndex, backingIndexEqualTo(dataStreamName, 2));
             assertThat(dsBackingIndices.get(0), is(thirtySecondsDownsampleIndex));
         }, 30, TimeUnit.SECONDS);
+        assertDownsamplingMethod(downsamplingMethod, thirtySecondsDownsampleIndex);
+    }
+
+    /**
+     * This test ensures that when we change the sampling method, the already downsampled indices will use the original sampling method,
+     * while the raw data ones will be downsampled with the most recent configuration.
```

Comment:
Perhaps you were already planning this, but I'd recommend adding some docs somewhere that describe this update behavior. I don't know downsampling that well, so TBH I have trouble understanding this test (i.e. why the `5m` index gets deleted), but that's probably on me.

Reply:
I agree, my concern is that we need to see documentation more holistically, not only on the API level. I will add a note there, but the sampling method needs to be documented and explained properly.

Reply:
There is a documentation  task in the meta issue: https://github.com/elastic/elasticsearch/issues/128357